sources/packages/api/src/route-controllers/application/application.controller.service.ts

```diff
@@ -203,4 +209,76 @@ export class ApplicationControllerService {
       );
     }
   }
+
+  /**
+   * Process application income verification details.
+   * @param CRAIncomeVerification CRA income verification data.
+   * @return processed object.
+   */
+  processApplicationIncomeVerificationDetails(
+    CRAIncomeVerification: CRAIncomeVerification[],
+  ): ApplicationIncomeVerification {
+    const incomeVerificationDetails = {} as ApplicationIncomeVerification;
+    const studentIncomeVerifications = CRAIncomeVerification.filter(
+      (incomeVerification) => !incomeVerification.supportingUser,
+    );
+    const supportingUserIncomeVerification = CRAIncomeVerification.filter(
+      (incomeVerification) => incomeVerification.supportingUser,
+    );
+    const [studentIncomeVerification] = studentIncomeVerifications;
+    // Student income verification details.
+    incomeVerificationDetails.studentIncomeVerificationStatus = {
+      waiting: !studentIncomeVerification.dateReceived,
+      success: !!studentIncomeVerification.dateReceived,
```

Comment:
Also as a status which is derived from if dateReceived is present, completed is more suitable than success IMO.